src/TesApi.Web/BatchPool.cs

```diff
@@ -0,0 +1,584 @@
+        {
+            var exceptions = new List<Exception>();
+
+            await PerformTask(ServicePoolAsync(IBatchPool.ServiceKind.GetResizeErrors, cancellationToken), cancellationToken);
+            await PerformTask(ServicePoolAsync(IBatchPool.ServiceKind.ManagePoolScaling, cancellationToken), cancellationToken);
+            await PerformTask(ServicePoolAsync(IBatchPool.ServiceKind.Rotate, cancellationToken), cancellationToken);
+            await PerformTask(ServicePoolAsync(IBatchPool.ServiceKind.RemovePoolIfEmpty, cancellationToken), cancellationToken);
+
+            switch (exceptions.Count)
+            {
+                case 0:
+                    return;
+
+                case 1:
+                    throw exceptions.First();
+
+                default:
+                    throw new AggregateException(exceptions.SelectMany(Flatten).ToArray());
+            }
+
+            static IEnumerable<Exception> Flatten(Exception ex)
+                => ex switch
+                {
+                    AggregateException aggregateException => aggregateException.InnerExceptions,
+                    _ => Enumerable.Empty<Exception>().Append(ex),
+                };
+
+            async ValueTask PerformTask(ValueTask serviceAction, CancellationToken cancellationToken)
+            {
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    try
+                    {
+                        await serviceAction;
+                    }
+                    catch (Exception ex)
+                    {
+                        exceptions.Add(ex);
+                        RemoveMissingPools(ex);
+                    }
+                }
+            }
+
+            void RemoveMissingPools(Exception ex)
+            {
+                switch (ex)
+                {
+                    case AggregateException aggregateException:
+                        foreach (var e in aggregateException.InnerExceptions)
+                        {
+                            RemoveMissingPools(e);
+                        }
+                        break;
+
+                    case BatchException batchException:
+                        if (batchException.RequestInformation.BatchError.Code == BatchErrorCodeStrings.PoolNotFound)
+                        {
+                            _logger.LogError(ex, "Batch pool {PoolId} is missing. Removing it from TES's active pool list.", Pool.PoolId);
+                            _ = _batchPools.RemovePoolFromList(this);
+                        }
+                        break;
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<DateTime> GetAllocationStateTransitionTime(CancellationToken cancellationToken = default) // TODO: put this at front of list by returning earliest possible time, or put at end of list by returning UtcNow?
```

Comment:
Forgot to remove the comment after I reached the determination. `AllocationStateTransitionTime` is what we want, and if that doesn't exist (for whatever reason), put this at the end of the line (because the only reasonable case for that property being null is because the pool is really new.